odxtools/communicationparameter.py

```diff
@@ -103,7 +106,7 @@ def short_name(self):
         # This should not happen anyway in a correct PDX
         return self.id_ref.ref_id.replace(".", "__").replace("-", "_")
 
-def read_communication_param_ref_from_odx(et_element, doc_frags: List[OdxDocFragment]):
+def read_communication_param_ref_from_odx(et_element, doc_frags: List[OdxDocFragment], dl_variant_type): # TODO argument type annotation
```

Comment:
please add the type annotation for `dl_variant_type` ;)

I suppose it is
```suggestion
def read_communication_param_ref_from_odx(et_element, doc_frags: List[OdxDocFragment], dl_variant_type: str):
```

Reply:
Good catch! I would even go a step further and make this string an Enum type, since it also occurs at other places in the code.

Reply:
go ahead. I repeatedly fell on my nose when trying to use `Literal` for such things, but if you can make it work without too much hassle, I'm all for it...

Reply:
The latest version implements a `DIAG_LAYER_TYPE` enum to distinguish the types of diag layers by a concrete type. The `__init__` method of the class `DiagLayer` accepts only enum instances, which ensures correctness-by-construction.